File: lab2_task3/main.swift
func countSequences(_ sequence: [Int]) -> Int {
    guard sequence.count >= 2 else { return 0 }
    
    var seqLen = 1
    var seqCount = 0
    
    for i in 0..<sequence.count - 1 {
        if sequence[i+1] >= sequence[i] {
            seqLen += 1
        } else if seqLen > 1 {
            seqCount += 1
            seqLen = 1
        } else {
            seqLen = 1
        }
    }
    
    if seqLen > 1 {
        seqCount += 1
    }
    
    return seqCount
}

print("Введите последовательность чисел через пробел:")
if let input = readLine() {
    let nums = input.split(separator: " ").compactMap { Int($0) }
    print(countSequences(nums))
}

File: lab2_task3/main.cs
using System;
using System.Collections.Generic;
using System.Linq;

class Program {
    static int CountSequences(List<int> sequence) {
        if (sequence.Count < 2) return 0;
        
        int seqLen = 1;
        int seqCount = 0;
        
        for (int i = 0; i < sequence.Count - 1; i++) {
            if (sequence[i+1] >= sequence[i]) {
                seqLen++;
            }
            else if (seqLen > 1) {
                seqCount++;
                seqLen = 1;
            }
            else {
                seqLen = 1;
            }
        }
        
        if (seqLen > 1) seqCount++;
        
        return seqCount;
    }

    static void Main() {
        Console.WriteLine("Введите последовательность чисел через пробел:");
        List<int> nums = Console.ReadLine().Split()
                          .Select(int.Parse).ToList();
        Console.WriteLine(CountSequences(nums));
    }
}